lib/utils/html_widgets.dart

```diff
@@ -0,0 +1,152 @@
+        CustomHtmlWidgets.BUTTON =>
+          _buildCustomButtonWidget(element, resources, eventTriggers),
+        CustomHtmlWidgets.VIDEO =>
+          _buildCustomVideoWidget(element, resources, eventTriggers),
+        _ => null
+      };
+
+  /// Builds custom Image widget from HTML [element] and applies [resources]
+  ///
+  /// TODO: Implement loading images from device memory
+  static Widget _buildCustomImageWidget(
+    dom.Element element,
+    List<ExhibitionPageResource> resources,
+    List<ExhibitionPageEventTrigger> eventTriggers,
+  ) {
+    Size size = _extractSize(element);
+    ExhibitionPageEventTrigger? foundEventTrigger =
+        _findEventTrigger(element, eventTriggers);
+    String? foundResource = resources
+        .firstWhereOrNull((resource) => resource.id == element.innerHtml)
+        ?.data;
+
+    Widget imageWidget = Image.network(
+      foundResource ?? "",
+      width: size.width,
+      height: size.height,
+    );
+
+    if (foundEventTrigger != null) {
+      return GestureDetector(
+        onTap: _handleTapEvent(foundEventTrigger),
+        child: imageWidget,
+      );
+    }
+
+    return imageWidget;
+  }
+
+  /// Builds custom Button widget from HTML [element] and applies [resources]
+  static Widget _buildCustomButtonWidget(
+    dom.Element element,
+    List<ExhibitionPageResource> resources,
+    List<ExhibitionPageEventTrigger> eventTriggers,
+  ) {
+    Size size = _extractSize(element);
+    ExhibitionPageEventTrigger? foundEventTrigger =
+        _findEventTrigger(element, eventTriggers);
+    String? foundResource = resources
+        .firstWhereOrNull((resource) => resource.id == element.innerHtml)
+        ?.data;
+
+    return ElevatedButton(
+        style: ButtonStyle(fixedSize: MaterialStateProperty.all(size)),
+        onPressed: foundEventTrigger != null
+            ? _handleTapEvent(foundEventTrigger)
+            : null,
+        child: Text(foundResource ?? ""));
+  }
+
+  /// Builds custom Video widget from HTML [element] and applies [resources]
+  static Widget? _buildCustomVideoWidget(
+    dom.Element element,
+    List<ExhibitionPageResource> resources,
+    List<ExhibitionPageEventTrigger> eventTriggers,
+  ) {
+    Size size = _extractSize(element);
+    ExhibitionPageEventTrigger? foundEventTrigger =
+        _findEventTrigger(element, eventTriggers);
+    String? source = element.children
+        .firstWhereOrNull((element) => element.localName == "source")
+        ?.attributes[CustomHtmlWidgets.SRC];
+    String? foundResource =
+        resources.firstWhereOrNull((resource) => resource.id == source)?.data;
+
+    if (source != null) {
+      return Container(
+        width: size.width,
+        height: size.height,
+        child: VideoWidget(src: foundResource ?? ""),
+      );
+    }
+
+    return null;
+  }
+
+  /// Finds event trigger assigned to this [element] from [eventTriggers]
+  static ExhibitionPageEventTrigger? _findEventTrigger(
```

Comment:
This finds only clickView triggers, so it should be renamed to match the behaviour, e.g. findClickViewEventTrigger

Reply:
Resolved